docs/advanced/classes.rst

```diff
@@ -1259,3 +1259,40 @@ object, just like ``type(ob)`` in Python.
     Other types, like ``py::type::of<int>()``, do not work, see :ref:`type-conversions`.
 
 .. versionadded:: 2.6
+
+Custom type setup
+=================
+
+For advanced use cases, such as setting certain type object slots, you may wish
+to directly manipulate the `PyHeapTypeObject` corresponding to a ``py::class_``
+definition.
+
+You can do that using ``py::custom_type_setup``:
+
+.. code-block:: cpp
```

Comment:
I'm really worried about this very large & complex block of code in the documentation. Bit rot seems almost guaranteed. Could this be moved to the test, with a simple pointer here? (I wish we had an examples directory in this repo, but that'd be a project on its own.)

Reply:
This is also in a test.  But it seemed useful to provide a working example.  I was able to simplify it slightly since I also fixed `object::operator=`.

Reply:
I still feel strongly that
* the code duplication by itself is worrisome.
* the potential for bit rot (untested code) is worrisome.
* the large code block in the documentation gives too much weight to a feature that I think will be fairly esoteric to most pybind11 users.

Reply:
FYI, I'm fine if someone wants to setup testing for our docs, I think it would be extremely useful. Probably by moving the examples to files, then using inclusion markers to pull it into the docs, but maybe the other direction (making files out of the docs then testing those) is possible too. But that should be a separate project.